src/grt/src/fastroute/src/utility.cpp

```diff
@@ -519,8 +537,13 @@ void FastRouteCore::assignEdge(int netID, int edgeID, bool processDIR)
       for (l = 0; l < num_layers_; l++) {
         if (layer_grid_[l][k - 1] > 0) {
           gridD[l][k - 1] = gridD[l][k] + 1;
-        } else {
+        } else if (layer_grid_[l][k] == 0) {
           gridD[l][k - 1] = gridD[l][k] + BIG_INT;
+        } else {
+          // when the layer orientation doesn't match the edge orientation,
+          // set a larger weight to avoid assigning to this layer when the routing
+          // has 3D overflow
+          gridD[l][k - 1] = gridD[l][k] + 2*BIG_INT;
```

Comment:
When would this happen?  Going wrong way across a gcell is odd.

Reply:
In my congestion unit tests, I've noticed that layer assignment can assign an edge to a layer without respecting the preferred direction. It happens because layer assignment looks only to the (capacity - usage) value. Wrong-way will have this value equal to zero, so I've added a larger weight to this scenario to avoid the wrong way assignment.

Reply:
I've also fixed the way I was detecting wrong-way assignments. Previously, I was looking only at negative values in layer_grid_, but we can have negative values in edges with the correct direction. I changed it by assigning std::numeric_limits<int>::min() for wrong-way layers.

Reply:
can we just drop wrong way edges from the graph?

Reply:
I'm not sure if I understand your question.
My update in the `assignEdge` function is removing the wrong-way layers from the possible layers for an edge, so we'll never have an edge assigned to a wrong-way layer.

Reply:
You are raising the cost rather than removing its existence.  That's good enough but slightly different.